### ISSUE TITLE
Bug Alert in Logs

### ISSUE THREAD

**Issue by Daxcor69** [projectcontour/contour#6385]
[2024-04-28 13:34:08.862][1][error][envoy_bug] [source/common/runtime/runtime_impl.cc:501] envoy bug failure: false. Details: Using a removed guard envoy.reloadable_features.sanitize_te. In future version of Envoy this will be treated as invalid configuration
[2024-04-28 13:34:08.862][1][error][envoy_bug] [./source/common/common/assert.h:38] stacktrace for envoy bug
[2024-04-28 13:34:08.862][1][error][envoy_bug] [./source/common/common/assert.h:45] #0 UNKNOWN [0x55d70b98c09b]
[2024-04-28 13:34:08.862][1][error][envoy_bug] [./source/common/common/assert.h:45] #1 UNKNOWN [0x55d70b990059]
[2024-04-28 13:34:08.862][1][error][envoy_bug] [./source/common/common/assert.h:45] #2 UNKNOWN [0x55d70b98de43]
[2024-04-28 13:34:08.862][1][error][envoy_bug] [./source/common/common/assert.h:45] #3 UNKNOWN [0x55d70b98f4ba]
[2024-04-28 13:34:08.862][1][error][envoy_bug] [./source/common/common/assert.h:45] #4 UNKNOWN [0x55d70b06046d]
[2024-04-28 13:34:08.862][1][error][envoy_bug] [./source/common/common/assert.h:45] #5 UNKNOWN [0x55d70b060b02]
[2024-04-28 13:34:08.862][1][error][envoy_bug] [./source/common/common/assert.h:45] #6 UNKNOWN [0x55d70b065e8a]
[2024-04-28 13:34:08.862][1][error][envoy_bug] [./source/common/common/assert.h:45] #7 UNKNOWN [0x55d70b067baa]
[2024-04-28 13:34:08.862][1][error][envoy_bug] [./source/common/common/assert.h:45] #8 UNKNOWN [0x55d70b069b91]
[2024-04-28 13:34:08.862][1][error][envoy_bug] [./source/common/common/assert.h:45] #9 UNKNOWN [0x55d70b595585]
[2024-04-28 13:34:08.862][1][error][envoy_bug] [./source/common/common/assert.h:45] #10 UNKNOWN [0x55d70b59c5dd]
[2024-04-28 13:34:08.862][1][error][envoy_bug] [./source/common/common/assert.h:45] #11 UNKNOWN [0x55d70b781b0d]
[2024-04-28 13:34:08.862][1][error][envoy_bug] [./source/common/common/assert.h:45] #12 UNKNOWN [0x55d70b806233]
[2024-04-28 13:34:08.862][1][error][envoy_bug] [./source/common/common/assert.h:45] #13 UNKNOWN [0x55d70b4d52f2]
[2024-04-28 13:34:08.862][1][error][envoy_bug] [./source/common/common/assert.h:45] #14 UNKNOWN [0x55d70b6fe280]
[2024-04-28 13:34:08.863][1][error][envoy_bug] [./source/common/common/assert.h:45] #15 UNKNOWN [0x55d70b70a2fb]

kubernetes 1.29.1
contour:
envoy:
os: ubuntu 22.04
kubeadm install
helm: 17.0.7

Do I need to be concerned with this error.  Here is the over ride values to helm I used.

```
contour:
  nodeSelector:
envoy:
  kind: daemonset
  nodeSelector:
    node-role.kubernetes.io/proxy: ""
  logLevel: info
  service:
    type: LoadBalancer
    externalTrafficPolicy: Local
    loadBalancerIP: "x.x.x.x"
```

**Comment by skriss** [comment]
Which versions of contour + envoy are you running? Contour 1.28.2 and 1.28.3 set this runtime setting (see https://github.com/projectcontour/contour/blob/main/changelogs/CHANGELOG-v1.28.2.md for info), but should be using Envoy 1.29.2 and 1.29.3, respectively, which both support the runtime flag. My guess is you have an unsupported Contour + Envoy version combo.

**Comment by Daxcor69** [comment]
CHART NAME: contour
CHART VERSION: 17.0.7
APP VERSION: 1.28.3

This is from the helm chart. I will looks to see if there is a newer version of the helm chart

**Comment by tsaarni** [comment]
You can run:

```
helm install my-release oci://registry-1.docker.io/bitnamicharts/contour --set envoy.image.tag=1.29.3-debian-12-r1
```

personally I do not know how the versions end up in the bitnami helm chart.

**Comment by skriss** [comment]
> Personally I do not know how the versions end up in the bitnami helm chart https://github.com/bitnami/charts/pull/25117

Huh yeah looks like there have been unsupported version combos in there for quite awhile (we do try to maintain compatibility so probably not usually an issue), possibly worth filing an issue in that repo about it.

**Comment by tsaarni** [comment]
Fixed in chart version 17.0.8
 https://github.com/bitnami/charts/pull/25494
